### ISSUE TITLE
PointRend Masks are not going upto edge

### ISSUE THREAD

**Issue by Swaroop11** [facebookresearch/detectron2#1352]
Hi All,

I was training the PointRend project on my custom dataset, 30K train images (Each image had 2-3 instances in them) using 16GPUs (Tesla K80). I made the training script using train_net.py inside the PointRend folder.

Training went completely fine and used all 16 GPUs for 5K steps, I used learning rate scheduler and warmup steps also, and loss was also reducing. 

After training, when I start testing the model it gave me random masks for some of the instances. 
1.  In some images, masks were really good, but in others, masks are not going up to edges of instance, the mask is only up to half of the object from one side.

2. There was no consistency with these results like it was not happening for a particular category of instance, nor dependent on the size of the instance.

I need help to know is there any issue with my training process or any other change in config file is required for better mask results.

Thanks and Regard
Swaroop

**Comment by ppwwyyxx** [comment]
As the [issue template](https://github.com/facebookresearch/detectron2/blob/master/.github/ISSUE_TEMPLATE/unexpected-problems-bugs.md) mentions:

> If you expect the model to converge / work better, note that we do not give suggestions on how to train a new model. Only in one of the two conditions we will help with it: (1) You're unable to reproduce the results in detectron2 model zoo. (2) It indicates a detectron2 bug.

We provide configs & models with standard academic settings and expect users to have the knowledge to choose or design appropriate models & parameters for their own tasks.

**Comment by Swaroop11** [comment]
I understand your point, but I need to know whether this issue is because of training on 16GPUs and not properly managing the learning rate scheduler or some other mistake I am making.

When I trained Mask RCNN ResNet50 with the same data but on 1 GPU masks came out to be very good.  No issue with the incomplete mask. 

But with PointRend and 16 GPUs this issue is coming up, can you help me with this???

**Comment by bpfliegel** [comment]
@Swaroop11 Swaroop, did you find the root cause? I think I am experiencing the same as you had - masks are frequently 'cut' from single or multiple sides.
Thanks for any insights.

**Comment by bpfliegel** [comment]
> @bpfliegel did you find something useful to remove the issue??

No, but did not spend any major effort on it. I think PointRend itself is excellent (if you adjust its sampling based on the Cityscapes config, it performs really really well) - I think cropping is also present without PointRend (well it's always present, right? tight/relaxed predictions are natural), but I have to check source codes if there could be any remedy. Like relaxing the instances at a stage and let PointRend do its magic - I'm bluffing, have to get to know Detectron internals well.

If you find any remedy or good alternative, kindly let me know, will do the same.
